department-of-veterans-affairs/caseflow
# JusKha/APPEALS-29405

Resolves [APPEALS-29405 Create Intake Form Shell](https://jira.devops.va.gov/browse/APPEALS-29405)

# Description
Created a shell for the intake correspondence queue. Created a progress bar to follow progress throughout the intake. Created feature tests for the current functionality. 

## Acceptance Criteria
- [ ] The "Intake Form" has the below multi stage form headers:
1. Add Related Correspondence
- The form is defaulted to "Add Related Correspondence"
2. Review Tasks & Appeals
3. Confirm

- [ ] There is a bar visible under the multi step headers
1. If the user is on step 2, both step 1 and 2 are styled black
2. The remaining steps are styled gray

- [ ] A Continue button is visible on "Add Related Correspondence" and "Review Tasks & Appeals" steps
- When the Continue button is clicked the user goes to the next step

- [ ] A Back button is visible on the "Review Tasks & Appeals" and "Confirm" steps
- When the Back button is clicked the user goes to the previous step

- [ ] A Submit button is visible on the "Confirm" step

- [ ] A Cancel link is visible on all steps 
- When the Cancel link is clicked, the user is directed to the Correspondence Queue

- [ ] If the feature flag :correspondence_queue is disabled, then the user is routed to /unauthorized

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

1. Log in as any user (such as BVAAABSHIRE)
2. In the address bar, navigate to /queue/correspondence/12345/intake
3. Verify the progress bar appears with the first bar active. Also verify only continue and cancel buttons visible
4. Click on the Continue button. Verify the progress bar moves and the back button appears.
5. Click on the Continue button again. Verify the progress bar moves and the continue button becomes a submit button.
6. Click on the Cancel button. Verify the URL matches /queue/correspondence, and the screen will say "Page not found"

# Frontend
## User Facing Changes
 
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/4c37f9fd-8e5c-4768-a4e4-29447c47337e)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/0bef8a14-61b1-4dea-b230-f2797c94005a)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/19611dde-9903-44ca-85a9-d12de27ea970)
